graylog2-server/src/main/java/org/graylog/security/shares/EntitySharesService.java

```diff
@@ -98,32 +98,32 @@ public EntitySharePrepareResponse prepareShare(GRN ownedEntity,
      * The grants in the request are created or, if they already exist, updated.
      *
      * @param ownedEntity the target entity for the updated grants
-     * @param request     the request containing grantees and their roles
+     * @param request     the request containing grantees and their capabilities
      * @param sharingUser the user executing the request
      */
     public EntitySharePrepareResponse updateEntityShares(GRN ownedEntity, EntityShareRequest request, User sharingUser) {
         final String userName = sharingUser.getName();
-        final List<GrantDTO> existingGrants = grantService.getForTarget(ownedEntity, grnRegistry.newGRN("user", sharingUser.getName()));
```

Comment:
did we reach a decision regarding this?
In my book we were still filtering the current user..
This also needs to be handled by the frontend now. (If they don't send back your own grant, it will get deleted)

Reply:
Good point. I actually saw this as a bug. :smile: My understanding was, that we need to allow that a user can remove its own ownership for an entity. That means the sharing user needs to be in the active shares and shown in the list of users in the share dialog. If we exclude the current user from the active shares, we need a different way for allowing a user to remove its own ownership.

Reply:
From a UX perspective, I'd say this will clutter the UI, because it's a rare condition.
But we can keep the API generic. The frontend could still hide it..